src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -34,21 +35,31 @@ internal class AttributeCloner<TAttribute>
         // Compute the values to apply to Settable properties on newly created attribute.
         private readonly Action<TAttribute, BindingData>[] _propertySetters;
 
-        // Optional hook for post-processing the attribute. This is intended for legacy hack rules.
-        private readonly Func<TAttribute, Task<TAttribute>> _hook;
-
         private readonly Dictionary<PropertyInfo, AutoResolveAttribute> _autoResolves = new Dictionary<PropertyInfo, AutoResolveAttribute>();
 
         private static readonly BindingFlags Flags = BindingFlags.Instance | BindingFlags.Public;
 
+        private readonly string _methodName;
+
+        public AttributeCloner(
+            TAttribute source,
+            BindingDataContract bindingDataContract,
+            INameResolver nameResolver) : this(source, bindingDataContract, ClonerContext.New(nameResolver))
+        {
+        }
+
         public AttributeCloner(
             TAttribute source,
             BindingDataContract bindingDataContract,
-            INameResolver nameResolver = null,
-            Func<TAttribute, Task<TAttribute>> hook = null)
+            ClonerContext context = null)
         {
-            _hook = hook;
-            nameResolver = nameResolver ?? new EmptyNameResolver();
+            if (context == null)
+            {
+                context = new ClonerContext();
+            }
+            _methodName = context.MethodName;
```

Comment:
recommend a field for the context so as the context grows over time we don't have to copy individual values off?